src/components/wallet/wallet.js

```diff
@@ -121,22 +125,55 @@ export default class Wallet extends Component {
                             {
                                 method_name: "ft_transfer_call",
                                 args: Base64.encode(JSON.stringify({
-                                    receiver_id: STORAGE.addresses.multicall, 
+                                    receiver_id: multicall, 
                                     amount: amount,
                                     msg: JSON.stringify({
                                         function_id: "multicall",
                                         args: Base64.encode(JSON.stringify({"calls":LAYOUT.toBase64()}).toString())
                                     }).toString()
                                 })),
-                                deposit: "1", // nep-141 ft_transfer_call gets EXACTLY 1 yocto
-                                gas: `${gas}`
+                                deposit: "1", // nep-141 specifies EXACTLY 1 yocto
+                                gas: gas
                             }
                         ]
                     }
                 }
             }
         }
 
+        // check if multicall has enough storage on Token
+        const [storageBalance, storageBounds] = await Promise.all([
+            // get storage balance of multicall on the token
+            view(
+                token,
+                "storage_balance_of",
+                { account_id: multicall }
+            ).catch(e => "0"), // return 0 if failed
+            // get storage balance bounds in case multicall has no storage on the token and it needs to be paid
+            view(
+                token,
+                "storage_balance_bounds",
+                {}
+            ).catch(e => {})
+        ]);
+        const totalStorageBalance = Big(storageBalance?.total ?? "0");
+        const storageMinBound = Big(storageBounds.min);
```

Comment:
You put `?.` in the previous line, so just to make sure, is `storageBounds.min` always defined or should it also have `?.`?